logging/dedupe.go

```diff
@@ -9,6 +9,43 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+const (
```

Comment:
seems this change should be part of the preceding commit.

Reply:
Fixed!  Lots of rebasing to try and make this all a little easier to review.